src/modules/Channel/context/hooks/useSendFileMessageCallback.ts

```diff
@@ -28,169 +28,60 @@ export default function useSendFileMessageCallback(
   { logger, pubSub, scrollRef, messagesDispatcher }: UseSendFileMessageCallbackParams,
 ) {
   const sendMessage = useCallback(
-    (file: File, quoteMessage = null) => new Promise<FileMessage>((resolve, reject) => {
-      const { compressionRate, resizingWidth, resizingHeight } = imageCompression ?? {};
-      const shouldCreateCustomParams = onBeforeSendFileMessage && typeof onBeforeSendFileMessage === 'function';
-
-      const compressibleFileType = file.type === 'image/jpg' || file.type === 'image/png' || file.type === 'image/jpeg';
-      const compressibleRatio = compressionRate > 0 && compressionRate < 1;
-      // pxToNumber returns null if values are invalid
-      const compressibleDiamensions = utils.pxToNumber(resizingWidth) || utils.pxToNumber(resizingHeight);
-
-      const canCompressImage = compressibleFileType && (compressibleRatio || compressibleDiamensions);
-
-      const createParamsDefault = (fileToSend: File) => {
-        const params: FileMessageCreateParams = {
-          file: fileToSend,
-        };
+    (compressedFile: File, quoteMessage = null) => new Promise<FileMessage>((resolve, reject) => {
+      // Create FileMessageParams
+      let params: FileMessageCreateParams = onBeforeSendFileMessage?.(compressedFile, quoteMessage);
+      if (!params) {
+        params = { file: compressedFile };
         if (quoteMessage) {
           params.isReplyToChannel = true;
           params.parentMessageId = quoteMessage.messageId;
         }
```

Comment:
Q. Why doesn't the params created for onBeforeSendFileMessage handle the following processing?
```
if (quoteMessage) {
  params.isReplyToChannel = true;
  params.parentMessageId = quoteMessage.messageId;
}
```

Reply:
It was not applying this process to the onBeforeSendFileMessage before.
The onBeforeSendFileMessage is a function that totally depends on the customer's input